File: _src/05/Chapter5.VB/Context.vb
Imports System.Data.Entity
Imports System.Data.Entity.ModelConfiguration.Conventions
Imports Chapter5.VB.Models
Imports Chapter5.VB.Maps

Public Class Context
    Inherits DbContext
    Public Sub New()
        MyBase.New("name=chapter5")
    End Sub
    Property People As DbSet(Of Person)
    Property Phones() As DbSet(Of Phone)
    Property PersonTypes() As DbSet(Of PersonType)
    Property Companies() As DbSet(Of Company)

    Protected Overrides Sub OnModelCreating(ByVal modelBuilder As DbModelBuilder)
        modelBuilder.Configurations.Add(New PersonMap)
        modelBuilder.Configurations.Add(New PersonTypeMap)
        modelBuilder.Configurations.Add(New CompanyMap)
        modelBuilder.Configurations.Add(New AddressMap)
    End Sub

End Class

File: _src/05/Chapter5.CSharp/Context.cs
using System.Data.Entity;
using System.Data.Entity.ModelConfiguration.Conventions;
using Chapter5.CSharp.Maps;
using Chapter5.CSharp.Models;

namespace Chapter5.CSharp
{
    public class Context : DbContext
    {
        public Context()
            : base("name=Chapter5")
        {

        }
        public DbSet<Person> People { get; set; }
        public DbSet<Phone> Phones { get; set; }
        public DbSet<PersonType> PersonTypes { get; set; }
        public DbSet<Company> Companies { get; set; }
        protected override void OnModelCreating(DbModelBuilder modelBuilder)
        {
            modelBuilder.Configurations.Add(new PersonMap());
            modelBuilder.Configurations.Add(new PersonTypeMap());
            modelBuilder.Configurations.Add(new CompanyMap());
            modelBuilder.Configurations.Add(new AddressMap());
        }

    }
}
